core/threaded/reactor_threaded.c

```diff
@@ -1072,8 +1072,13 @@ int lf_reactor_c_main(int argc, const char* argv[]) {
 
   LF_PRINT_DEBUG("Start time: " PRINTF_TIME "ns", start_time);
   struct timespec physical_time_timespec = {start_time / BILLION, start_time % BILLION};
+
+#ifdef NO_TTY
+  lf_print("---- Start execution ----");
```

Comment:
I'm pretty sure I think it was supposed to mean "we have no standard output" or "do not print anything." Perhaps `NO_STDOUT` would be a better name.

Reply:
Tagging @edwardalee and @erlingrj to draw their attention to this issue.

Here is a candidate proposal:
 - rename `NO_TTY` to `NO_STDOUT` and make sure that all print statements are guarded by it
 - add a `NO_CTIME` definition to filter out any calls to `ctime`

Thoughts?

Reply:
One of the original goals of having lf_print and related functions was to be able to redefine them for embedded platforms, for example to suppress stdout when it is not available.  The #ifdef should therefore be inside the lf_print definition, not outside, shouldn't it?

Reply:
Originally the NO_TTY definition was meant to capture that this is an embedded target where you probably don't have a terminal interface to the program. So e.g. passing command line args doesn't make sense. We probably don't want to disable printing, because that is still common on embedded targets, e.g. routing it out to an UART where there might or might not be something connected.

This particular problem is that we don't want to use ctime because it is expensive. We could also consider a define e.g. NO_STD which indicates that we don't have a full-blown standard library and will avoid such calls? Or maybe MINIMAL_STD

Reply:
Hmm, maybe `NO_CLI` (command line interface) would be a better naming for not being able to pass in command line arguments? Then we could have three macros: `NO_TTY` which means no printing whatsoever, `NO_CLI` which means no command line interface and finally `MINIMAL_STDLIB` which means we filter out unecessary calls like `ctime`.

Reply:
@magnmaeh sounds like a good proposal to me. Curious to hear what the other think.

Reply:
@edwardalee  and @erlingrj?

Reply:
I am fine with renaming NO_TTY to NO_CLI, and introducing MINIMAL_STDLIB and use it here. I dont think this PR should introduce this new version of NO_TTY as it requires some more thought.